server.go

```diff
@@ -208,6 +212,25 @@ func (server *Server) handle(s *streamWrap) error {
 func (s *service) svcCall(sWrap *streamWrap, mtype *methodType, svcID ServiceID, ctxv, argv, replyv reflect.Value) error {
 	function := mtype.method.Func
 
+	startT := time.Now()
+	var cctx context.Context
+	var ok bool
+	cctx, ok = ctxv.Interface().(context.Context)
```

Comment:
Why not doing this code block inside `handle()` where you have the original `ctx`? That way you wouldn't need to re-cast into the original type.

Perhaps worth separating into it's own helper function.